net_udp.go

```diff
@@ -147,6 +147,33 @@ func newNetUDPSummary(file string) (*NetUDPSummary, error) {
 	return netUDPSummary, nil
 }
 
+// NOTE: this function assumes the architecture is little endian, like x86 and amd64
```

Comment:
I'm not really comfortable assuming little endian here unless the kernel is consistent about the file format.

I have systems with big endian (mips64).

Reply:
This is what chatgpt recommends, seems reasonable: https://chat.openai.com/share/2586e0df-70b5-4b54-a55a-29a60e4b689d

```golang
func IsLittleEndian() bool {
    var testValue uint16 = 0xABCD
    testBytes := make([]byte, 2)
    binary.LittleEndian.PutUint16(testBytes, testValue)
    return testBytes[0] == 0xCD
}
```